FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/abstractions/platform/freertos/transport_interface_freertos.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* Standard includes. */
+#include <string.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "atomic.h"
+#include "semphr.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_Sockets.h"
+
+/* Transport interface include. */
+#include "transport_interface_freertos.h"
+
+struct NetworkContext
+{
+    Socket_t tcpSocket;
+};
+
+BaseType_t Transport_FreeRTOS_Connect( NetworkContext_t * pNetworkContext,
+                                       const char * pHostName,
+                                       uint16_t port,
+                                       uint32_t receiveTimeoutMs )
+{
+    Socket_t tcpSocket = FREERTOS_INVALID_SOCKET;
+    BaseType_t socketStatus = 0;
+    struct freertos_sockaddr serverAddress = { 0 };
+    TickType_t receiveTimeout = pdMS_TO_TICKS( receiveTimeoutMs );
```

Comment:
As this code belongs to FreeRTOS/FreeRTOS, I think it is a good idea to follow the naming conventions of FreeRTOS, but this code is placed under FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/, so see a good point to support either the current one or FreeRTOS naming convention. What do you think?

Reply:
After bringing it up with Richard, the kernel naming convention is preferred, but he is fine with either one. I refrained from making the change at the moment, since I thought it might be a little odd if the parameter names in the transport send/receive implementations didn't match up with the names in the CSDK where the function pointers are typedef'ed. The naming change can also be in a follow up PR if we want it.

Reply:
Ok sounds good. Thanks